src/lib/plugins/tp.js

```diff
@@ -25,6 +25,11 @@ module.exports.server = (serv) => {
         let y = serv.posFromString(args[1], ctx.player.position.y)
         let z = serv.posFromString(args[2], ctx.player.position.z)
 
+        if (x > 29999999 || y > 4096 || y < -4096 || z > 29999999) {
```

Comment:
What about negative x and z coordinates? You can check for the absolute values of x y and z instead.